external-crates/move/crates/move-package/src/lib.rs

```diff
@@ -186,7 +187,9 @@ impl BuildConfig {
             lock_string,
         )?;
 
-        if modified {
+        if modified || install_dir_set {
+            // (1) Write the Move.lock file if the existing one is `modified`, or
+            // (2) `install_dir` is set explicitly, which may be a different directory, and where a Move.lock does not exist yet.
```

Comment:
This addresses what I consider a bug: Previously, if we set `install_dir` to something other than the current directory (let's say `/tmp/foo`) then if we ran build on the directory, and the `Move.lock` did not change, then the `Move.lock` file would not be written to `install_dir`. Now, we write `Move.lock` to `install_dir` if it is explicitly set.

I think it's clear the intention is to put `Move.lock` in the `install_dir` (because, if no `Move.lock` file exists, and we build it, then we do place it in the `install_dir`). Indeed our tests rely on checking the `Move.lock` contents in `install_dir`, which is convenient.

Reply:
Reasonable!  We should arguably be also looking in the install dir for the lock file that `DependencyGraph` reads as well, to completely fix this bug, but the difference in behaviour in what you have here vs that is likely an un-exercised edge case, so all good.